browser_history/generic.py

```diff
@@ -302,20 +324,83 @@ def fetch_bookmarks(self, bookmarks_paths=None, sort=True, desc=False):
                 output_object.bookmarks.sort(reverse=desc)
         return output_object
 
+    def fetch_cookies(self, cookies_paths=None, sort=True, desc=False):
+        """Returns cookies of all available profiles stored in SQL.
+        The returned datetimes are timezone-aware with the local timezone set
+        by default.
+        The cookies files are first copied to a temporary location and then
+        queried, this might lead to some additional overhead and results
+        returned might not be the latest if the browser is in use. This is
+        done because the SQlite files are locked by the browser when in use.
+        :param cookies_paths: (optional) a list of cookie files.
+        :type cookies_paths: list(:py:class:`pathlib.Path`)
+        :param sort: (optional) flag to specify if the output should be
+            sorted. Default value set to True.
+        :type sort: boolean
+        :param desc: (optional)  flag to specify asc/desc
+            (Applicable if sort is True) Default value set to False.
+        :type asc: boolean
+        :return: Object of class :py:class:`browser_history.generic.Outputs`
+            with the data member cookies set to
+            list(tuple(:py:class: str, str, str, `datetime.datetime`, str, str)).
+            If the browser is not installed, this object will be empty.
+        :rtype: :py:class:`browser_history.generic.Outputs`
+        """
+        if cookies_paths is None:
+            cookies_paths = self.paths(profile_file=self.cookies_file)
+        output_object = Outputs(fetch_type="cookies")
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            for cookies_path in cookies_paths:
+                copied_cookies_path = shutil.copy2(cookies_path.absolute(), tmpdirname)
+                conn = sqlite3.connect(f"file:{copied_cookies_path}?mode=ro", uri=True)
+                cursor = conn.cursor()
+                cursor.execute(self.cookies_SQL)
+                cookies = [
+                    {
+                        "name": name,
+                        "url": host + path,
+                        "encrypted_value": e_value,
+                        "value": value,
+                        "expiry": d,
+                        "isSecure": "Secure" if isSecure == 1 else "Not Secure",
+                        "isHTTP": "HTTP" if isHttpsOnly == 1 else "Not HTTP",
+                    }
+                    for (
+                        name,
+                        host,
+                        path,
+                        e_value,
+                        value,
+                        d,
+                        isSecure,
+                        isHttpsOnly,
+                    ) in cursor.fetchall()
+                ]
+                date_cookies = []
+                for cookie in cookies:
+                    date_cookies.append(self.cookies_parser(cookie))
```

Comment:
Continuing from the other comment, you should skip cookies that couldn't be parsed here.